pygsti/modelmembers/states/densestate.py

```diff
@@ -265,13 +265,14 @@ def __init__(self, purevec, basis, evotype, state_space):
             else _statespace.StateSpace.cast(state_space)
         evotype = _Evotype.cast(evotype)
         basis = _Basis.cast(basis, state_space.dim)  # basis for Hilbert-Schmidt (superop) space
-
-        #Try to create a dense pure rep.  If this fails, see if a dense superkey rep
+        
+        #Try to create a dense pure rep.  If this fails, see if a dense superket rep
         # can be created, as this type of rep can also hold arbitrary pure states.
         try:
             rep = evotype.create_pure_state_rep(purevec, basis, state_space)
             self._reptype = 'pure'
-            self._purevec = self._basis = None
+            self._purevec = None
+            self._basis = basis #this was previously being set as None, not sure why.
```

Comment:
I think perhaps self._basis isn't used.  I should be fine to set it, and makes sense to do so.